prow/plugins/lint/lint.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lint
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/golang/lint"
+
+	"k8s.io/test-infra/prow/git"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "lint"
+
+var lintRe = regexp.MustCompile(`(?mi)^/lint\s*$`)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIC)
+}
+
+type githubClient interface {
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+	CreateComment(org, repo string, number int, body string) error
+}
+
+func handleIC(pc plugins.PluginClient, ic github.IssueCommentEvent) error {
+	return handle(pc.GitHubClient, pc.GitClient, pc.Logger, ic)
+}
+
+// modifiedGoFiles returns a map from filename to patch string for all go files
+// that are modified in the PR.
+func modifiedGoFiles(ghc githubClient, org, repo string, number int) (map[string]string, error) {
+	changes, err := ghc.GetPullRequestChanges(org, repo, number)
+	if err != nil {
+		return nil, err
+	}
+
+	modifiedFiles := make(map[string]string)
+	for _, change := range changes {
+		if filepath.Ext(change.Filename) == ".go" {
+			modifiedFiles[change.Filename] = change.Patch
+		}
+	}
+	return modifiedFiles, nil
+}
+
+// problemsInPatches runs golint on the files. It returns all lint problems
+// that are in added lines in the files' patches.
```

Comment:
what if the patch deletes a function and that causes lint to fail in other places?